components/rendering/src/markdown.rs

```diff
@@ -1,20 +1,121 @@
 use std::borrow::Cow::Owned;
+use std::iter::Peekable;
 
 use pulldown_cmark as cmark;
 use self::cmark::{Parser, Event, Tag, Options, OPTION_ENABLE_TABLES, OPTION_ENABLE_FOOTNOTES};
 use slug::slugify;
 use syntect::easy::HighlightLines;
 use syntect::html::{start_coloured_html_snippet, styles_to_coloured_html, IncludeBackground};
 
-use errors::Result;
+use errors::{Result, Error};
 use utils::site::resolve_internal_link;
 use context::Context;
 use highlighting::{SYNTAX_SET, THEME_SET};
 use short_code::{SHORTCODE_RE, ShortCode, parse_shortcode, render_simple_shortcode};
 use table_of_contents::{TempHeader, Header, make_table_of_contents};
 
+struct GutenbergFlavoredMarkdownParser<'a, 'b, 'c> {
+    context: &'a Context<'a>,
+    errors: &'c mut Vec<Error>,
+    parser: Peekable<Parser<'b>>,
+}
+
+fn is_internal_link(link: &str) -> bool {
```

Comment:
this is only used once so I don't think it needs to be a function

Reply:
Yes, it's only used once, but I think the value here is being explicit about what the parsing is looking for. When simply using `link.starts_with("./")` I think it's not as apparent that what we're really looking for is an internal link.